example-utils/usubscription-static-file/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+/********************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+#![allow(clippy::mutable_key_type)]
+
+use async_trait::async_trait;
+use serde_json::Value;
+use std::collections::HashSet;
+use std::fs::{self, canonicalize};
+use std::path::PathBuf;
+use std::str::FromStr;
+use up_rust::core::usubscription::{
+    FetchSubscribersRequest, FetchSubscribersResponse, FetchSubscriptionsRequest,
+    FetchSubscriptionsResponse, NotificationsRequest, SubscriberInfo, Subscription,
+    SubscriptionRequest, SubscriptionResponse, USubscription, UnsubscribeRequest,
+};
+use up_rust::{UStatus, UUri};
+
+pub struct USubscriptionStaticFile {
+    static_file: PathBuf,
+}
+
+impl USubscriptionStaticFile {
+    pub fn new(static_file: Option<PathBuf>) -> Self {
+        // Set static_file to static_file if it has value, else set to "static-configs/testdata.json"
+        let static_file =
+            static_file.unwrap_or_else(|| PathBuf::from("static-configs/testdata.json"));
+        USubscriptionStaticFile { static_file }
+    }
+}
+
+#[async_trait]
+impl USubscription for USubscriptionStaticFile {
```

Comment:
Now that I better understand how we'd need to submit Request messages through to the uSubscription service, I suppose this would change completely, right?

We'd have to do something like:
* configure which is the host transport
* use the host transport to submit
  * 1 x register for notifications request and then
  * 1 x request for all subscriptions

Would you agree? If so, I'll go open an issue to implement this.

Tagging @stevenhartley on this for visibility.

Reply:
yes, I think that makes sense. Do we register for notifications on all topics? Or like, topics for which we have forwarders?

Reply:
We'd register for notifications for all topics.

Reply:
Opened an issue to track this: #36